java-checks/src/main/java/org/sonar/java/checks/ReplaceLambdaByMethodRefCheck.java

```diff
@@ -54,7 +60,7 @@ public void scanFile(JavaFileScannerContext context) {
 
   @Override
   public void visitLambdaExpression(LambdaExpressionTree tree) {
-    if (isMethodInvocation(tree.body()) || isBlockInvokingMethod(tree.body())) {
+    if ((isMethodInvocation(tree.body()) && isUsingLambdaParamAsArg(tree)) || isBlockInvokingMethod(tree.body())) {
```

Comment:
@m-g-sonar I would move `isMethodInvocation(tree.body())` inside `isUsingLambdaParamAsArg` method because right now the cast to `MethodInvocationTree` happens inside `isUsingLambdaParamAsArg` which makes the rule a little unclear.